### ISSUE TITLE
Error: internal error, response body doesn't match error type signature

### ISSUE THREAD

**Issue by arbourd** [s1ntaxe770r/terraform-provider-planetscale#10]
Appears that the response body isn't matching what's returned. In my cause it's probably related to a 401 or something re: access because everything looks good but returns the same error.

```hcl
resource "planetscale_branch" "db" {
  organization  = "company"
  database      = "comapny-db"
  parent_branch = "main"
  name          = var.name == "production" ? "main" : var.name
}
```

**Comment by arbourd** [comment]
Appears the error is related to the `planetscale-go` library:
https://github.com/planetscale/planetscale-go/blob/7adddf4d6d4888abc4fa77141e9facc09f32a8f7/planetscale/client.go#L196-L211

**Comment by arbourd** [comment]
Seems like this happens when the API token is incorrect. Probably returning a 401 which isn't handled properly by the client library.

**Comment by arbourd** [comment]
Caused by upstream planetscale/planetscale-go#118

**Comment by s1ntaxe770r** [comment]
resolved in #11

**Comment by arbourd** [comment]
@s1ntaxe770r It wasn't actually that was just updating the library hoping it would be. It's an upstream bug with the `planetscale-go` library.
